internal/pkg/agent/vault/vault_notdarwin.go

```diff
@@ -0,0 +1,214 @@
+}
+
+// Set stores the key in the vault store
+func (v *Vault) Set(ctx context.Context, key string, data []byte) (err error) {
+	enc, err := v.encrypt(data)
+	if err != nil {
+		return err
+	}
+
+	err = v.tryLock(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = v.unlock(err)
+	}()
+
+	return writeFile(v.filepathFromKey(key), enc)
+}
+
+// Get retrieves the key from the vault store
+func (v *Vault) Get(ctx context.Context, key string) (dec []byte, err error) {
+	err = v.tryRLock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = v.unlock(err)
+	}()
+
+	enc, err := os.ReadFile(v.filepathFromKey(key))
+	if err != nil {
+		return nil, err
+	}
+
+	return v.decrypt(enc)
+}
+
+// Exists checks if the key exists
+func (v *Vault) Exists(ctx context.Context, key string) (ok bool, err error) {
+	err = v.tryRLock(ctx)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		err = v.unlock(err)
+	}()
+
+	if _, err = os.Stat(v.filepathFromKey(key)); err == nil {
+		ok = true
+	} else if errors.Is(err, fs.ErrNotExist) {
+		err = nil
+	}
+	return ok, err
+}
+
+// Remove removes the key
+func (v *Vault) Remove(ctx context.Context, key string) (err error) {
+	err = v.tryLock(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = v.unlock(err)
+	}()
+
+	return os.RemoveAll(v.filepathFromKey(key))
+}
+
+// Close closes the vault store
+// Noop for non-darwin implementation
+func (v *Vault) Close() error {
+	return nil
+}
+
+// applyOptions applies options for windows and linux, not used for darwin implementation
+func applyOptions(opts ...OptionFunc) Options {
+	options := Options{
+		lockRetryDelay: defaultFlockRetryDelay,
+	}
+
+	for _, opt := range opts {
+		opt(&options)
+	}
+
+	return options
+}
+
+// fileNameFromKey returns the filename as a hash of the vault seed combined with the key
+// this ties the key with the vault seed eliminating the change of attempting
+// to decrypt the key for the wrong vault seed value.
+func fileNameFromKey(seed []byte, key string) string {
+	hash := sha256.Sum256(append(seed, []byte(key)...))
+	return hex.EncodeToString(hash[:])
+}
+
+func (v *Vault) filepathFromKey(key string) string {
+	return filepath.Join(v.path, fileNameFromKey(v.seed, key))
+}
+
+// try to acquire exclusive lock
+func (v *Vault) tryLock(ctx context.Context) error {
+	_, err := v.lock.TryLockContext(ctx, v.lockRetryDelay)
+	if err != nil {
+		return fmt.Errorf("failed to acquire exclusive lock: %v, err: %w", v.lock.Path(), err)
+	}
+	return nil
+}
+
+// try to acquire shared lock
+func (v *Vault) tryRLock(ctx context.Context) error {
+	_, err := v.lock.TryRLockContext(ctx, v.lockRetryDelay)
+	if err != nil {
+		return fmt.Errorf("failed to acquire shared lock: %v, err: %w", v.lock.Path(), err)
+	}
+	return nil
+}
+
+// unlock unlocks the file lock and preserves the original error if there was a error
+func (v *Vault) unlock(err error) error {
+	return errors.Join(err, v.lock.Unlock())
+}
```

Comment:
[Blocker]
I don't understand why an unlock function needs to receive an error. The caller is responsible to handle any previous error and then decide what to do if `unlock` fails as well.

Besides it harms the single responsibility principle, unlock is also responsible to aggregate an error for no obvious reason.

Reply:
do not agree with making it a blocker, it's a helper function to make the deferred call code above more compact and consistent, since it's done in multiple places